### PR TITLE
[TableGen] Use consistent field kind checking in getNumericKey.

### PR DESCRIPTION
**Pull request by nvjle**
Fields GenericField::IsInstruction and GenericField::Enum can be simultaneously active for a given field.  Methods compareBy and primaryRepresentation both order the checking of IsInstruction before GenericField::Enum. Do the same in getNumericKey for consistency.

### CODE REVIEW

**Comment by nvjle** [comment]
Hi @piotrAMD , I hope you could review this tiny patch as discussed in #80486 .

File: llvm/utils/TableGen/SearchableTableEmitter.cpp [hunk 1]
```diff
@@ -215,12 +215,14 @@ int64_t SearchableTableEmitter::getNumericKey(const SearchIndex &Index,
...
-  if (Index.Fields[0].Enum) {
-    Record *EnumEntry = Rec->getValueAsDef(Index.Fields[0].Name);
-    return Index.Fields[0].Enum->EntryMap[EnumEntry]->second;
-  } else if (Index.Fields[0].IsInstruction) {
+  // To be consistent with compareBy and primaryRepresentation elsewhere,
+  // we check for IsInstruction before Enum-- these fields are not exclusive.
+  if (Index.Fields[0].IsInstruction) {
     Record *TheDef = Rec->getValueAsDef(Index.Fields[0].Name);
     return Target->getInstrIntValue(TheDef);
+  } else if (Index.Fields[0].Enum) {
```
Comment (piotrAMD) [inline llvm/utils/TableGen/SearchableTableEmitter.cpp:223]: Nit: No need for `else if` after return, `if` is enough.

**Review by piotrAMD — APPROVED** [review]
Thanks for the quick fix!

Normally it would need a test case, but I wasn't able to come up with a reproducer myself and I realize it can be tricky.

File: llvm/utils/TableGen/SearchableTableEmitter.cpp (hunk 1 above)
Comment (nvjle) [reply llvm/utils/TableGen/SearchableTableEmitter.cpp]: Done.
Could I trouble you or @wangpc-pp to merge on my behalf? Thank you.

**Pull request merged**